hikari/api/rest.py

```diff
@@ -96,6 +96,20 @@ class RESTClient(traits.NetworkSettingsAware, abc.ABC):
 
     __slots__: typing.Sequence[str] = ()
 
+    @property
+    @abc.abstractmethod
+    def token(self) -> typing.Union[str, TokenStrategy, None]:
+        """Token this client is using for most requests.
+
+        Returns
+        -------
+        typing.Union[builtins.str, TokenStrategy, builtins.None]
+            The token this client is using for most requests.
+
+            If this is `builtins.None` then this client will only work for some
+            endpoints such as public and webhook ones.
+        """
```

Comment:
I dont think this is a good thing to expose in any way

Reply:
Why not.

Reply:
Tokens are private and not something that should be in the public interface. I know we already store it in `_token`, but that is private internal detail, not something that we should expose

Reply:
Why shouldn't they be exposed and if a REST client doesn't have a token it can return None here

Reply:
Still not sure exactly what you're getting at with that private stuff but if you think keeping tokens "private" in this language is going to help with security you're just wrong, most people will be targetting the main implementation and it doesn't take a genius to just do `rest._token`. This doesn't change the fact that people can access the token, this just adds a more formalised way for people who want to introspect it.
Plus fwiw this is the only Discord API lib i've seen that doesn't have a way to get the rest client's active token